### PR TITLE
Solve incompatibility errors in _compression.py

### PR DESCRIPTION
**Pull request by ygalvao**
Increase compatibility across different virtual environments, particularly with Conda, when using brotli.Decompressor()

### CODE REVIEW

**Comment by wRAR** [comment]
Is this only required when `brotlipy` is installed instead of/in addition to `brotli`?

**Comment by ygalvao** [comment]
Yes. I don't know why, but when setting up conda environments and installing scrapy and other packages it seems Conda / Mamba installs only brotlipy.

**Comment by Gallaecio** [comment]
Maybe we should do something to make Scrapy distinguish the 2, and only enable brotli support when the right one is installed.

**Comment by wRAR** [comment]
Yeah.

**Comment by wRAR** [comment]
> when setting up conda environments and installing scrapy and other packages it seems Conda / Mamba only installs brotlipy.

Note that Scrapy currently does not depend on brotli (or on brotlipy), neither on PyPI nor on Conda, and if something installs brotlipy as opposed to not installing anything it's either "other packages" you mentioned or some other things you do when "setting up".

**Comment by ygalvao** [comment]
I understand. However, in my current conda env, I just have a few packages installed, most of them well-known ones like Scrapy, Selenium, Pandas, Numpy, etc. and somehow, Conda or Mamba installed brotlipy, which throws an AttributeError when running Scrapy. So this simple change solves this issue, adding compatibility with brotlipy and hence with other packages installed using Conda / Mamba as well.

File: scrapy/utils/_compression.py [hunk 1]
```diff
@@ -5,6 +5,19 @@
     import brotli
 except ImportError:
     pass
+else:
+    try:
+        brotli.Decompressor.process
+    except AttributeError:
+
+        def _brotli_decompress(decompressor, data):
```
Comment (wRAR) [inline scrapy/utils/_compression.py:13]: Should we add a warning here?

By the way `brotlipy` is now `brotlicffi` (with the import name adjusted accordingly), but whether we need to add support for it is a separate question.

Comment (Gallaecio) [reply scrapy/utils/_compression.py:13]: [Sounds like support may make sense for PyPy support](https://github.com/urllib3/urllib3/issues/2040), but indeed out of scope here.

**Review by wRAR — APPROVED** [review]

**Pull request merged**